### PR TITLE
SCUMM: Allow detecting games from disabled subengines

### PR DESCRIPTION
**Pull request by ccawley2011**

### CODE REVIEW

**Comment by aquadran** [comment]
that makes things more simple :)

**Review by criezy — APPROVED** [review]
My understanding is that one of the goal of the refactoring done during GSoC was to allow detection of games even when an engine is not enabled. So it makes sense to also detect games for disabled sub-engines, whether the main engine is enabled or not.

**Comment by bluegr** [comment]
I also agree, since the detection code is split from the engine code, we may now detect all supported games.

One future enhancement here (out of scope of this PR) could be to show game entries gray in the ScummVM launcher, when an entry refers to a sub-engine that hasn't been compiled in the executable.

**Review by bluegr — APPROVED** [review]

**Comment by sev-** [comment]
Indeed! I'll also take a look at other similar cases in other engines.

File: engines/scumm/detection.cpp [hunk 1]
```diff
@@ -130,23 +130,7 @@ DetectedGames ScummMetaEngineDetection::detectGames(const Common::FSList &fslist
 }
 
 const char *ScummMetaEngineDetection::getName() const {
-	return "SCUMM ["
```
Comment (sev-) [inline engines/scumm/detection.cpp:133]: However, this change should not be here. I'll revert it. This shows which subengines were compiled in

**Pull request merged**